Lib/test/test_subprocess.py

```diff
@@ -1360,6 +1361,25 @@ def test_communicate_epipe(self):
         self.addCleanup(p.stdin.close)
         p.communicate(b"x" * 2**20)
 
+    def test_string_representation(self):
+        """Test string representation Popen object."""
+        code = 'import sys; sys.exit(57)'
+        cmd = [sys.executable, '-c', code]
+        proc = subprocess.Popen(cmd)
+
+        args = ' '.join(map(shlex.quote, cmd))
+        if len(args) > 30:
+            args = f"{args[:30]}...".rstrip()
+
+        unfinished = (
+            f"<subprocess.Popen: pid:{proc.pid} "
+            f"args:{args} returncode:unfinished>"
+        )
+        self.assertEqual(repr(proc), unfinished)
```

Comment:
This seems like it could suffer from timing issues, and we should avoid tests which fail only sometimes, even very rarely.

I suggest either of the following:

1. Use a dummy object instead of a real `Popen` instance, and just call `Popen.__repr__()` on it.
2. Add a `time.sleep(0.05)` call to the code, and move the proc creation to just before the call to repr(proc).

I prefer the first option since it avoids timing issues, the need to actually run a process, the need to import shlex, etc.

Reply:
Oh, a dummy object kind of sucks, it'll be much better to get the confidence of using a real `Popen` object. How about a process that awaits for input and then finishes? Then there are no timing issues.

Reply:
While I generally agree that exercise real code is better than mocks/dummies/stubs, since this is just testing the repr, I think a dummy object would be fine here.

> How about a process that awaits for input and then finishes?

That would be fine :)